### PR TITLE
MNT-21481: IMAP UID search command errors with French Locale- Only folders showing

### PR DESCRIPTION
**Pull request by amohammedalfresco**
- Fixed Outlook client connection issues with IMAP Server
- Added a Unit test to mock Outlook locale issue with IMAP
- Updated file headers for `Copyright (C) 2005 - 2020 Alfresco Software Limited`
- Added project to Travis CI system

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# the cache can grow constantly
+before_cache:
+  - rm -rf $HOME/.m2/repository/org/alfresco/alfresco-greenmail
+
+branches:
+  only:
+    - master
+    - /support\/.*/
+    - /feature\/.*/
+    - /fix\/.*/
```
Comment (killerboot) [inline .travis.yml:22]: I don't think it is required to build all these branches. Especially not for this project. Please leave "master" and "support", the rest will be built as PRs.

File: src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java [hunk 2]
```diff
@@ -299,7 +300,7 @@ private static SearchTermBuilder createDateSearchTermBuilder(final int compariso
             public SearchTerm build() {
                 String dateStr = getStringParameter(0);
                 try {
-                    Date date = new SimpleDateFormat("dd-MMM-yyyy").parse(dateStr);
+                    Date date = new SimpleDateFormat("dd-MMM-yyyy", Locale.ENGLISH).parse(dateStr);
```
Comment (killerboot) [inline src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java:303]: How does this work in French locale? Is the date format the same? What about other locales where date format is different (mm/dd/yy for example)?

File: .travis.yml (hunk 1 above)
Comment (amohammedalfresco) [reply .travis.yml]: Sure.

File: src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java (hunk 2 above)
Comment (amohammedalfresco) [reply src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java:303]: I believe it now sets `Locale.ENGLISH` as a base filter and parse it for `ReceivedDateTerm` or `SentDateTerm` methods.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
```
Comment (killerboot) [inline src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java:5]: The date is wrong, should be 2020.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class SearchTermBuilderTest {
```
Comment (killerboot) [inline src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java:37]: Sorry, but you are not testing SearchTermBuilder class here.
Currently you only check if SimpleDateFormat works, which we shouldn't check as it is a 3rd party class.

File: .travis.settings.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<settings>
+    <servers>
+        <server>
+            <id>alfresco-public</id>
```
Comment (killerboot) [inline .travis.settings.xml:4]: The id of the server does not match the id in the pom's distributionManagement config. I believe the release job will not work.

File: src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java (hunk 2 above)
Comment (amohammedalfresco) [reply src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java:303]: Re-wrote the Class/Method accordingly: https://github.com/Alfresco/alfresco-greenmail/blob/763e4ecab4ebc7ed21fce9bed3094cec6a816182/src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java#L33-L43

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java (hunk 3 above)
Comment (amohammedalfresco) [reply src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java]: Fixed for all the files.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java (hunk 4 above)
Comment (amohammedalfresco) [reply src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java:31]: Re-wrote the Class/Method accordingly: https://github.com/Alfresco/alfresco-greenmail/blob/763e4ecab4ebc7ed21fce9bed3094cec6a816182/src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java#L33-L43

File: src/main/java/com/icegreen/greenmail/AbstractServer.java [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
- * 
- * You should have received a copy of the GNU Lesser General Public License
- * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
- * #L%
- */
+/*
+ * #%L
+ * Alfresco greenmail implementation
+ * %%
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
```
Comment (killerboot) [inline src/main/java/com/icegreen/greenmail/AbstractServer.java:5]: Please revert the changes in the header files. The date needs to be updated only in the classes that were modified in this PR.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void testCreateDateSearchTermWithNonEnglishLocale()
+    {
+        Locale defaultLocale = Locale.getDefault();
+        Locale.setDefault(Locale.FRENCH);
+        SearchTermBuilder searchTermBuilder = SearchTermBuilder.create(SearchKey.SINCE.name());
+        String dateStr = "12-Jul-2020";
+        searchTermBuilder.addParameter(dateStr);
+        searchTermBuilder.build();
+        Locale.setDefault(defaultLocale);
```
Comment (killerboot) [inline src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java:42]: The locale needs to be set back in a reliable way. Please wrap the test in try-finally bock.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+ */
+package com.icegreen.greenmail.imap.commands;
+
+import java.util.Locale;
+import org.junit.Test;
+
+public class SearchTermBuilderTest {
+
+    @Test
+    public void testCreateDateSearchTermWithNonEnglishLocale()
```
Comment (killerboot) [inline src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java:34]: Does the test fail if the fix is removed from SearchTermBuilder? Is it worth adding a negative test?

File: src/main/java/com/icegreen/greenmail/AbstractServer.java (hunk 6 above)
Comment (amohammedalfresco) [reply src/main/java/com/icegreen/greenmail/AbstractServer.java]: Thanks, I reverted back and only updated classes that were worked in the PR.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java (hunk 7 above)
Comment (amohammedalfresco) [reply src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java]: Sure.  Thanks.  I've implemented it now.

File: src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java (hunk 8 above)
Comment (amohammedalfresco) [reply src/test/java/com/icegreen/greenmail/imap/commands/SearchTermBuilderTest.java:36]: Yup, just added a false positive test.

**Comment by amohammedalfresco** [comment]
Just noticed the GitHub information: `We’ve detected the file has mixed line endings. When you commit changes we will normalize them to Windows-style (CRLF).`
